crates/supervisor/core/src/reorg/handler.rs

```diff
@@ -56,19 +56,23 @@ where
                 managed_node.clone(),
             );
 
-            let handle = tokio::spawn(async move { reorg_task.process_chain_reorg().await });
+            let chain_id = *chain_id;
+
+            let handle = tokio::spawn(async move {
+                let reorg_task = reorg_task.with_metrics();
+                let start_time = Instant::now();
+                let result = reorg_task.process_chain_reorg().await;
+                super::metrics::Metrics::record_l1_reorg_processing(chain_id, start_time, &result);
+                result
+            });
             handles.push(handle);
         }
 
         let results = future::join_all(handles).await;
-        let failed_chains = results.into_iter().filter(|result| result.is_err()).count();
-
-        if failed_chains > 0 {
-            warn!(
-                target: "supervisor::reorg_handler",
-                no_of_failed_chains = %failed_chains,
-                "Reorg processing completed with failed chains"
-            );
+        for result in results {
+            if let Err(err) = result {
+                error!(target: "supervisor::reorg_handler", %err, "Error processing reorg task");
```

Comment:
is this a problem with the supervisor program or the kona-node, or with the individual chains? can we pls match the error and emit warn vs error log respectively